grype/db/curator.go

```diff
@@ -214,7 +226,12 @@ func (c *Curator) UpdateTo(listing *ListingEntry, downloadProgress, importProgre
 
 // Validate checks the current database to ensure file integrity and if it can be used by this version of the application.
 func (c *Curator) Validate() error {
-	return c.validate(c.dbDir)
+	metadata, err := c.validate(c.dbDir)
+	if err != nil {
+		return err
+	}
+
+	return c.validateStaleness(metadata)
```

Comment:
it seems odd that `validateStaleness` is not part of the `validate` call -- if there is a specific reason it should probably be added as a comment here. Also if `validate` isn't covering all validations it should probably be renamed to the scope that it's responsible for, such as `validateIntegrity` or similar.

Reply:
Yes, I renamed it to `validateIntegrity`. That is the relevant check while importing a database for example.